esptool.py

```diff
@@ -380,34 +380,35 @@ def _connect_attempt(self, mode='default_reset', esp32r0_delay=False):
             time.sleep(0.05)
             self._port.setDTR(False)  # IO0=HIGH, done
 
-        for _ in range(5):
-            try:
-                self.flush_input()
-                self._port.flushOutput()
-                self.sync()
-                return None
-            except FatalError as e:
-                if esp32r0_delay:
-                    print('_', end='')
-                else:
-                    print('.', end='')
-                sys.stdout.flush()
-                time.sleep(0.05)
-                last_error = e
-        return last_error
+        if not no_sync:
```

Comment:
Rather than indenting all the sync logic, how about returning here if `no_sync` is set?

Reply:
Noted.